verilog/tools/ls/verilog-language-server_test.cc

```diff
@@ -0,0 +1,485 @@
+    SetRequest(request);
+    return ServerStep();
+  }
+
+  // Returns the latest responses from the Language Server
+  std::string GetResponse() {
+    std::string response = response_stream_.str();
+    response_stream_.str("");
+    response_stream_.clear();
+    return response;
+  }
+
+  // Returns response to textDocument/initialize request
+  std::string GetInitializeResponse() { return initialize_response_; }
+
+ private:
+  // Wraps request for the Language Server in RPC header
+  void SetRequest(absl::string_view request) {
+    request_stream_.clear();
+    request_stream_.str(
+        absl::StrCat("Content-Length: ", request.size(), "\r\n\r\n", request));
+  }
+
+  // Performs single VerilogLanguageServer step, fetching latest request
+  absl::Status ServerStep() {
+    return server_->Step([this](char *message, int size) -> int {
+      request_stream_.read(message, size);
+      return request_stream_.gcount();
+    });
+  }
+
+  // Sets up the testing environment - creates Language Server object and
+  // sends textDocument/initialize request.
+  // It stores the response in initialize_response field for further processing
+  void SetUp() override {
+    server_ = std::make_unique<VerilogLanguageServer>(
+        [this](absl::string_view response) { response_stream_ << response; });
+
+    absl::Status status = InitializeCommunication();
+    EXPECT_TRUE(status.ok()) << "Failed to read request:  " << status;
+    initialize_response_ = GetResponse();
+  }
+
+  // Currently tested instance of VerilogLanguageServer
+  std::unique_ptr<VerilogLanguageServer> server_;
+
+  // Response from textDocument/initialize request - left for checking e.g
+  // server capabilities
+  std::string initialize_response_;
+
+  // Stream for passing requests to the Language Server
+  std::stringstream request_stream_;
+
+  // Stream for receiving responses from the Language Server
+  std::stringstream response_stream_;
+};
+
+// Verifies textDocument/initialize request handling
+TEST_F(VerilogLanguageServerTest, InitializeRequest) {
+  std::string response_str = GetInitializeResponse();
+  json response = json::parse(response_str);
+
+  ASSERT_EQ(response["id"], 1) << "Response message ID invalid";
+  ASSERT_EQ(response["result"]["serverInfo"]["name"],
+            "Verible Verilog language server.")
+      << "Invalid Language Server name";
+}
+
+std::string DidOpenRequest(absl::string_view name, absl::string_view content) {
+  return absl::StrCat(
+      R"({ "jsonrpc": "2.0", "method": "textDocument/didOpen", "params": { "textDocument": {"uri": ")",
+      name, R"(", "text": ")", content, R"("}}})");
+}
+
+// Checks automatic diagnostics for opened file and textDocument/diagnostic
+// request for file with invalid syntax
+TEST_F(VerilogLanguageServerTest, SyntaxError) {
+  const std::string wrong_file =
+      DidOpenRequest("file://syntaxerror.sv", R"(brokenfile\n)");
+  absl::Status status = SendRequest(wrong_file);
```

Comment:
FYI, [in some tests](https://github.com/chipsalliance/verible/blob/master/common/util/file_util_test.cc#L30-L35), we often have a convenience macro `EXPECT_OK` and `ASSERT_OK` that allows to do checking in one line. Might be a cool thing in further changes (for the TODO list :) ).